docs/admin/config/batch_changes.mdx

```diff
@@ -409,3 +430,38 @@ batchSpecLibrary(first: 100, labels: ["featured"]) {
   }
 }
 ```
+
+### Variables
+
+<Callout type="note">Variables are supported in Sourcegraph v6.6 and more.</Callout>
+
+Templates use variables to make batch specs adaptable to different scenarios. When creating a template, you can define placeholders for values that will vary between different uses of the template. Users filling out the template see these variables as form fields where they can enter specific values like repository names, file paths, or commit messages.
+
+<Callout type="note">You cannot use a variable for the batch change name.</Callout>
+
+The `libraryVariables` field accepts an array of variable objects, each with the following configuration options:
+
+- **Name**: The variable identifier used in the template
+- **Display name**: Optional human-readable name shown in the form interface
+- **Pattern**: Regular expression for input validation
+- **Description**: Help text shown to users explaining what the variable is for
+- **Mandatory**: Boolean field indicating whether the variable is required
+- **Level**: Validation message severity level (`INFO`, `WARNING`, or `ERROR`)
+
+You can create templates by adding the `libraryVariables` field to the `createBatchSpecLibraryRecord` mutation. Here's an example that creates a simple template:
+
+```graphql
+createBatchSpecLibraryRecord(name:"Hello World Template",
+libraryVariables: [{
+    name:"REPOSITORY_QUERY",
+    displayName:"Repository Search Query",
+    pattern: ".+",
+    description:"The search query to find repositories.",
+    mandatory:true,
+    level:ERROR
+  }], spec: "version: 2\nname: hello-world\ndescription: Add Hello World to READMEs\n\non:\n  - repositoriesMatchingQuery: $REPOSITORY_QUERY\n\nsteps:\n  - run: echo Hello World | tee -a README.md\n    container: alpine:3\n\nchangesetTemplate:\n  title: Hello World\n  body: Add Hello World to README\n  branch: hello-world\n  commit:\n    message: Add Hello World to README") {
+    id
+  }
+```
+
+To update or remove variables from an existing template, you will need to recreate the entire batch spec library record using the `deleteBatchSpecLibraryRecord` and `createBatchSpecLibraryRecord` mutations.
```

Comment:
Something about the word "entire" here makes the process seem cumbersome, when I don't think it benefits us to position it that way. Maybe we should instead say "....existing template, you must recreate the batch spec library recording using..." .

Reply:
Good call!